Add --delegate-proxy option to endpoint activate

aaschaer opened the pull request:
Resolves #303 

Testing this was a bit of an ordeal.

First of all, it seems M2Crypto and pip don't get along on Ubuntu, so I was only able to confirm that optional dependency logic in setup.py / Makefile break on pip install. Fortunatly Ubuntu is aware of this issue and has a package `python-m2crypto` that can be installed with apt-get. This breaks our install via virtualenv model though...

Second of all, default GCS doesn't work with delegate_proxy activation. You need to use a Gridmap instead of the default MyProxyGridmapCallout for authorization. If anyone else wants to test this feel free to ask me for more details.

I did manage to manually activate and then ls an endpoint though, so this *is* working as far as I can tell. I will try to add some automated tests tomorrow, but I suspect that I won't be able to do much but confirm that the option is hidden if M2Crypto hasn't been imported unless I find a way around the pip incompatibility.

jaswilli commented:
Curious about the m2crypto issue, as I/we have installed it via pip before. Let's talk tomorrow.

aaschaer commented:
Well I'm happy enough with the code itself, but getting M2Crypto to work with Travis and Nose2 continues to be an issue. It seems to currently be in a state where pip claims that it has "Successfully installed M2Crypto-0.26.0" (other than the pypy builds, which fail on building the M2Crypto wheel) but the tests are skipping on M2Crypto import failures.

sirosen commented:
Do local runs of the testsuite work okay? I might consider punting on getting the Travis issues worked out until later, if we all can test locally.

Reviewing code changes now.

sirosen commented on adoc/endpoint_activate.1.adoc:
```diff
@@ -33,17 +35,31 @@ Myproxy activation requires your username and password for the myproxy
 
+NOTE: Delegate Proxy activation requires the optional dependency of
+M2Crypto. This dependency is optional as it frequently requires additional
+non-python packets to be installed on your machine. Consult 
```
Typo alert: "packets"

aaschaer commented:
Added a `--proxy-lifetime` option (also hidden unless M2Crypto imported) to configure the lifetime of the proxy being generated and set the default lifetime to 12 hours.

Changed "packets" to "packages" to fix my typo.

And yes, the tests run fine locally. When I run `make travis` in a virtualenv on my machine everything passes and the test that checks that `--delegate-proxy` is hidden if M2Crypto is not imported is skipped as expected.

sirosen commented on adoc/endpoint_activate.1.adoc:
```diff
@@ -33,17 +35,31 @@ Myproxy activation requires your username and password for the myproxy
+for details
+
+To use Delegate Proxy activation use the --delegate-proxy option with a
+file containing an 5.X09 certificate as an argument (e.g. an X.509
```
Typo: `5.X09`

aaschaer commented:
Pushed the copy-edits if we want to go ahead and merge this to get the release out. I should have the other edits done before too long though.
